### ISSUE TITLE
dev: deploy test environment once and use `Mutex` on TestSequencer to avoid race condition

### ISSUE THREAD

**Issue by ftupas** [kkrt-labs/kakarot-rpc#400]
Currently, a `KakarotTestEnvironment` is created for each test in [client.rs](https://github.com/kkrt-labs/kakarot-rpc/blob/main/crates/core/tests/client.rs). It would be nice to spin up a test environment using [`#[once]`](https://github.com/la10736/rstest#use-once-fixture) and share this across the test. Something like this

```rust
#[fixture]
#[once]
pub fn kakarot_test_env_ctx() -> Arc<KakarotTestEnvironmentContext> {
    block_on(async {
        let ctx = KakarotTestEnvironmentContext::setup().await;
        Arc::new(ctx)
    })
}

```

Initial work has been done to add [fixtures](https://github.com/ftupas/kakarot-rpc/blob/cd75d9efed9621c6ce97aa45f352abe6b35d1a12/crates/core/tests/fixtures.rs#L93-L99) using rstest in #399 . Ideally we'll have to add `Mutex` on the `TestSequencer` to avoid race conditions

**Comment by greged93** [comment]
Just adding a comment: maybe it is possible to only have a lock on the client, or even generate a new client for every test but have a lock when getting a nonce and sending the transaction (which would avoid sending transactions with incorrect nonces).

**Comment by Eikix** [comment]
What is the status of this?
Since @greged93 is working on a pr to load a katana instance from a dump, is this relevant?

**Comment by greged93** [comment]
Discussed with Fred, can be closed given #424.
